src/josepy/jwk.py

```diff
@@ -275,3 +252,130 @@ def fields_to_partial_json(self):
+                kwargs['key'], util.ComparableECKey):
+            kwargs['key'] = util.ComparableECKey(kwargs['key'])
+        super(JWKEC, self).__init__(*args, **kwargs)
+
+    @classmethod
+    def _encode_param(cls, data):
+        """Encode Base64urlUInt.
+        :type data: long
+        :rtype: unicode
+        """
+
+        def _leading_zeros(arg):
+            if len(arg) % 2:
+                return '0' + arg
+            return arg
+
+        return json_util.encode_b64jose(binascii.unhexlify(
+            _leading_zeros(hex(data)[2:].rstrip('L'))))
+
+    @classmethod
+    def _decode_param(cls, data, name, valid_lengths):
+        """Decode Base64urlUInt."""
+        try:
+            binary = json_util.decode_b64jose(data)
+            if len(binary) not in valid_lengths:
+                raise errors.DeserializationError(
+                    'Expected parameter "{name}" to be {valid_lengths} bytes '
+                    'after base64-decoding; got {length} bytes instead'.format(
+                        name=name, valid_lengths=valid_lengths, length=len(binary))
+                )
+            return int(binascii.hexlify(binary), 16)
+        except ValueError:  # invalid literal for long() with base 16
+            raise errors.DeserializationError()
+
+    @classmethod
+    def _curve_name_to_crv(cls, curve_name):
+        if curve_name == 'secp256r1':
+            return 'P-256'
+        if curve_name == 'secp384r1':
+            return 'P-384'
+        if curve_name == 'secp521r1':
+            return 'P-521'
+        raise errors.SerializationError()
+
+    @classmethod
+    def _crv_to_curve(cls, crv):
+        # crv is case-sensitive
+        if crv == 'P-256':
+            return ec.SECP256R1()
+        if crv == 'P-384':
+            return ec.SECP384R1()
+        if crv == 'P-521':
+            return ec.SECP521R1()
+        raise errors.DeserializationError()
+
+    @classmethod
+    def _expected_length_for_curve(cls, curve):
+        if isinstance(curve, ec.SECP256R1):
+            return range(32, 33)
+        elif isinstance(curve, ec.SECP384R1):
+            return range(48, 49)
+        elif isinstance(curve, ec.SECP521R1):
+            return range(63, 67)
+
+    def fields_to_partial_json(self):
+        params = {}
+        if isinstance(self.key._wrapped, ec.EllipticCurvePublicKey):
+            public = self.key.public_numbers()
+        elif isinstance(self.key._wrapped, ec.EllipticCurvePrivateKey):
+            private = self.key.private_numbers()
+            public = self.key.public_key().public_numbers()
+            params.update({
+                'd': private.private_value,
+            })
+        else:
+            raise errors.SerializationError(
+                'Supplied key is neither of type EllipticCurvePublicKey nor EllipticCurvePrivateKey')
+        params.update({
+            'x': public.x,
+            'y': public.y,
+        })
+        params = dict((key, self._encode_param(value))
```

Comment:
dict comprehensions are faster, and probably also more Pythonic. It works in Python 2.

`{key: self._encode_param(value) for key, value in six.iteritems(params)}`

Reply:
Thanks! Fixed by 6e8d041c